test/plugin/scenarios/rocketmq-5-grpc-scenario/src/main/java/test/apache/skywalking/apm/testcase/rocketmq/client/java/controller/CaseController.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.rocketmq.client.apis.ClientServiceProvider;
+import org.apache.rocketmq.client.apis.consumer.FilterExpression;
+import org.apache.rocketmq.client.apis.consumer.FilterExpressionType;
+import org.apache.rocketmq.client.apis.consumer.MessageListener;
+import org.apache.rocketmq.client.apis.consumer.ConsumeResult;
+import org.apache.rocketmq.client.apis.consumer.PushConsumer;
+import org.apache.rocketmq.client.apis.message.Message;
+import org.apache.rocketmq.client.apis.message.MessageView;
+import org.apache.rocketmq.client.apis.producer.Producer;
+import org.apache.rocketmq.client.apis.producer.SendReceipt;
+import org.apache.rocketmq.common.MixAll;
+import org.apache.rocketmq.tools.command.MQAdminStartup;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Date;
+
+@RestController
+@RequestMapping("/case")
+@Slf4j
+public class CaseController {
+
+    private static final String SUCCESS = "Success";
+
+    @Value("${endpoints}")
+    private String endpoints;
+
+    @Value("${nameServer}")
+    private String nameServer;
+
+    static final String TOPIC = "TopicTest";
+    static final String TAG = "TagA";
+    static final String GROUP = "group1";
+
+    @RequestMapping("/rocketmq-5-grpc-scenario")
+    @ResponseBody
+    public String testcase() {
+        try {
+            ClientServiceProvider provider = ClientServiceProvider.loadService();
+            ClientConfiguration clientConfiguration = ClientConfiguration.newBuilder()
+                    .setEndpoints(endpoints)
+                    .enableSsl(false)
+                    .build();
+            // start producer
+            Producer producer = provider.newProducerBuilder()
+                    .setClientConfiguration(clientConfiguration)
+                    .build();
+            System.out.printf("Provider Started.%n");
+
+            // send msg
+            Message message = provider.newMessageBuilder()
+                    // Set topic for the current message.
+                    .setTopic(TOPIC)
+                    // Message secondary classifier of message besides topic.
+                    .setTag(TAG)
+                    // Key(s) of the message, another way to mark message besides message id.
+                    .setKeys("KeyA")
+                    .setBody("This is a normal message for Apache RocketMQ".getBytes(StandardCharsets.UTF_8))
+                    .build();
+            SendReceipt sendReceipt = producer.send(message);
+            System.out.printf("%s send msg successfully, message: %s%n", new Date(), sendReceipt);
+
+            // start consumer
+            Thread thread = new Thread(new Runnable() {
```

Comment:
Is the provider and comsumer created duplicate for every request? This endpoint is called repeatedly when retry.

Reply:
I am a reference `rocketmq-scenario` demo

![image](https://github.com/apache/skywalking-java/assets/3367713/afec3f14-9c44-4c50-b1d2-f0b8a9108689)

Reply:
Ref can't prove it is correct or wrong. Sometimes, wrong things got missed from last review.